### ISSUE TITLE
No video

### ISSUE THREAD

**Issue by fdservices** [celluloid-player/celluloid#871]
Overview Description:
No video

Steps to Reproduce:
1. Open Celluloid
2. Select a file to play (mp4)
3. Screen is blank, progress meter moves across, audio plays

Versions:

Pinephone 1.2
PostmarketOS
Phosh
Linux pinephone 6.3.5 #1-postmarketos-allwinner SMP PREEMPT Tue Jun 13 06:24:32 UTC 20 aarch64 GNU/Linux

[celluloid.log](https://github.com/celluloid-player/celluloid/files/12206628/celluloid.log)

**Comment by gnome-mpv** [comment]
I can't reproduce this on Pinephone Pro, but I found a different problem. The video play fine, but somehow the phone turns off after about 3 seconds.

**Comment by fdservices** [comment]
When you say "turns off" do you mean suspends or actually powers off?

This may be a bit cheeky, but could you post your "world" contents. I have a feeling I am missing some software.

**Comment by gnome-mpv** [comment]
> When you say "turns off" do you mean suspends or actually powers off?

Actually, it looks like it doesn't actually power off. It looked as though it was going to power off because the boot splash showed up. But then it just went back to the lock screen.

> This may be a bit cheeky, but could you post your "world" contents. I have a feeling I am missing some software.

```
base 3-1
bootsplash-theme-danctnix 2-0
celluloid-git 0.25.r18.gc587612-1
chatty 0.7.3-2
cups 1:2.4.6-1
cups-filters 1.28.17-4
danctnix-phosh-ui-meta 0.2-3
danctnix-tweaks-app 0.12.0-2
danctnix-tweaks-app-phosh 0.12.0-2
danctnix-tweaks-app-pinephone 0.12.0-2
device-pine64-pinephonepro 0.4-1
dhcpcd 10.0.2-1
dialog 1:1.3_20230209-1
dosfstools 4.2-3
eog 44.3-1
evince-mobile 43.1-1
f2fs-tools 1.16.0-2
fwupd 1.9.3-1
geary 1:44.0-1
gedit 44.2-2
ghostscript 10.01.2-1
glibc-locales 2.33-5
gnome-calculator 44.0-1
gnome-calendar-mobile 44.1-1
gnome-calls 44.2-1
gnome-camera-git a13ec47-1
gnome-clocks-mobile 41.0-3
gnome-console 44.0-1
gnome-contacts-mobile 44.0-1
gnome-initial-setup-mobile 41.4-1
gnome-maps 44.3-1
gnome-software-mobile 43.1-1
gnome-sound-recorder 42.0-2
gnome-usage-mobile 3.38.1-2
gnome-weather 44.0-1
gsfonts 20200910-3
gst-plugin-gtk4 0.10.6-1
gst-plugins-bad 1.22.5-1
gst-plugins-good 1.22.5-1
gtherm 0.0.3-1
iw 5.19-2
lollypop 1.4.37-2
megapixels 1.6.1-1
mesa 23.1.4-2
mmsd-tng 2.0.0-2
mobile-config-firefox 4.0.1-1
net-tools 2.10-2
netctl 1.28-2
networkmanager 1.42.6-1
networkmanager-openvpn 1.10.2-3
noto-fonts-emoji 20220920-1
openssh 9.3p2-1
pipewire-alsa 1:0.3.76-2
pipewire-pulse 1:0.3.76-2
portfolio-file-manager 0.9.15-2
pv 1.7.0-1
screen 4.9.0-1
snapshot-git r229.5c2c4b4-1
sshfs 3.7.3-1
sudo 1.9.14.p3-1
v4l-utils 1.24.1-2
vi 1:070224-6
vim 9.0.1676-1
vvmplayer 2.2.0-1
which 2.21-6
wireless-regdb 2023.05.03-1
wpa_supplicant 2:2.10-8
xdg-user-dirs 0.18-1
youtube-dl 2021.12.17-2
zramswap 5-2
```

**Comment by fdservices** [comment]
Thanks for the listing. I will see if anything helps with my problem.

Try running with debug to see if anything is logged to explain the ppp suspending.

**Comment by fdservices** [comment]
No change I'm afraid.

In the log: I am not sure about these items:

`Applying default option --vo=opengl,vdpau,vaapi,xv,x11,libmpv,`

Should vo be opengl or opengles? Setting  the environment variable GDK_GL=gles makes no difference. x11? this is wayland. 

`celluloid-INFO: 10:06:15.787: Forcing --vo=null`
`celluloid-INFO: 10:06:15.811: Failed to retrieve property "current-vo" as string.`

I am assuming that vo is video out, in which case this is probably related to the opengl query above.

**Comment by gnome-mpv** [comment]
>In the log: I am not sure about these items:
>
>Applying default option --vo=opengl,vdpau,vaapi,xv,x11,libmpv,
>
>Should vo be opengl or opengles? Setting the environment variable GDK_GL=gles makes no difference. x11? this is wayland.

That's leftover code from back when X11 embedding was still supported. I'll remove it later. We set `vo=libmpv` after that point so it has no effect anyway.

>celluloid-INFO: 10:06:15.787: Forcing --vo=null
>celluloid-INFO: 10:06:15.811: Failed to retrieve property "current-vo" as string.
>
>I am assuming that vo is video out, in which case this is probably related to the opengl query above.

That's coming from the second mpv instance that we use to prefetch metadata for playlist items. It doesn't actually do any playback. There should a `Forcing --vo=libmpv` coming from the main instance somewhere else.

**Comment by fdservices** [comment]
> Forcing --vo=libmpv

Yes there is. So opengl opengles is no longer a problem?  Still no video

**Comment by maskedwitch** [comment]
This issue was re-introduced with 0.28. I've attached a log. The only thing that stuck out from a very quick skim was

> (io.github.celluloid_player.Celluloid:2): celluloid-CRITICAL **: 13:18:31.655: Failed to initialize render context celluloid-Message: 13:20:19.947: [vo/libmpv] No render context set.

[celluloid.log](https://github.com/user-attachments/files/19815267/celluloid.log)
